servant-server/src/Servant/Server/Internal.hs

```diff
@@ -10,11 +10,13 @@
 {-# LANGUAGE ScopedTypeVariables        #-}
 {-# LANGUAGE TypeFamilies               #-}
 {-# LANGUAGE TypeOperators              #-}
+{-# LANGUAGE UndecidableInstances       #-}
```

Comment:
I think this particular case is okay, but one big issue is that `UndecidableInstances` is a module pragma, and we won't notice anymore when new undecidable instances come in with this.